phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java

```diff
@@ -223,4 +252,67 @@ protected boolean getNextCoveredIndexRow(List<Cell> result) throws IOException {
         }
         return false;
     }
+
+    private Result getCDCImage(
+            Map<String, Object> preImageObj, Map<String, Object> changeImageObj,
+            boolean isIndexCellDeleteRow, Long indexCellTS, Cell firstCell,
+            boolean isChangeImageInScope, boolean isPreImageInScope, boolean isPostImageInScope) {
+        Map<String, Object> rowValueMap = new HashMap<>();
+
+        if (isPreImageInScope) {
+            rowValueMap.put(CDC_PRE_IMAGE, preImageObj);
+        }
+
+        if (isChangeImageInScope) {
+            rowValueMap.put(CDC_CHANGE_IMAGE, changeImageObj);
+        }
+
+        Map<String, Object> postImageObj = new HashMap<>();
+        if (isPostImageInScope) {
+            if (!isIndexCellDeleteRow) {
+                for (Map.Entry<String, Object> preImageObjCol : preImageObj.entrySet()) {
+                    postImageObj.put(preImageObjCol.getKey(), preImageObjCol.getValue());
+                }
+                for (Map.Entry<String, Object> changeImageObjCol : changeImageObj.entrySet()) {
+                    postImageObj.put(changeImageObjCol.getKey(), changeImageObjCol.getValue());
+                }
+            }
+            rowValueMap.put(CDC_POST_IMAGE, postImageObj);
+        }
+
+        if (isIndexCellDeleteRow) {
```

Comment:
We should not decide if the data table mutation is delete or upsert based on the index row mutation type. It should be determined based on the data table mutation type.